worlds/_bizhawk/README.md

```diff
@@ -0,0 +1,279 @@
+and docstrings to learn more about each function.
+
+```
+class ConnectionStatus
+class BizHawkContext
+
+class NotConnectedError
+class RequestFailedError
+class ConnectorError
+class SyncError
+
+async def read(ctx, read_list) -> list[bytes]
+async def write(ctx, write_list) -> None:
+async def guarded_read(ctx, read_list, guard_list) -> (list[bytes] | None)
+async def guarded_write(ctx, write_list, guard_list) -> bool
+
+async def lock(ctx) -> None
+async def unlock(ctx) -> None
+
+async def get_hash(ctx) -> str
+async def get_system(ctx) -> str
+async def get_cores(ctx) -> dict[str, str]
+async def ping(ctx) -> None
+
+async def display_message(ctx, message: str) -> None
+async def set_message_interval(ctx, value: float) -> None
+
+async def connect(ctx) -> bool
+def disconnect(ctx) -> None
+
+async def get_script_version(ctx) -> int
+async def send_requests(ctx, req_list) -> list[dict[str, Any]]
+```
+
+`send_requests` is what actually communicates with the connector, and any functions like `guarded_read` will build the
+requests and then call `send_requests` for you. You can call `send_requests` yourself for more direct control, but make
+sure to read the docs in `connector_bizhawk_generic.lua`.
+
+A bundle of requests sent by `send_requests` will all be executed on the same frame, and by extension so will any helper
+that calls `send_requests`. For example, if you were to call `read` with 3 items on your `read_list`, all 3 addresses
+will be read on the same frame and then sent back.
+
+It also means that by default the only way to run multiple requests on the same frame is for them to be included in the
+same `send_requests` call. As soon as the connector finishes responding to a list of requests, it will advance the frame
+before checking for the next batch.
+
+### Requests that depend on other requests
+
+The fact that you have to wait at least a frame to act on any response may raise concerns. For example, Pokemon
+Emerald's save data is at a dynamic location in memory. It moves around when you load a new map. There is a static
+variable that holds the address of the save data, so we want to read the static variable to get the save address, and
+then use that address in a `write` to send the player an item. But between the `read` that tells us the address of the
+save data and the `write` to save data itself, an arbitrary number of frames have executed, and the player may have
+loaded a new map, meaning we've written data to who knows where.
+
+There are two solutions to this problem.
+
+1. Use `guarded_write` instead of `write`. We can include a guard against the address changing, and the script will only
+perform the write if the data in memory matches what's in the guard. In the below example, `write_result` will be `True`
+if the guard validated and the data was written, and `False` if the guard failed to validate.
```

Comment:
This needs a preposition, though I don't know enough of the context/content to know which one to use. Most likely either "to" or "over"
```suggestion
if the guard validated and the data was written to, and `False` if the guard failed to validate.
```

Reply:
"The data" is the information you sent, getting written to an address, or to memory. The book was written. The data was written. Do you think it needs to be "was written to memory"?

Reply:
I would say either keep it as is or change it to "written to memory." Either one works in this context because the data itself is not being written to, it's the memory being written to.

Reply:
Ah, I see what this is describing now. Yeah, my first comment was wrong, whatever you want here is fine